lua/autorun/misc.lua

```diff
@@ -76,14 +77,22 @@ else
 
 	hook.Add("PlayerSelectSpawn","SpawnNextToPit",function(ply)
 		if ply.PitDeath then
-			local outsidespawns = {}
-			for k,v in pairs(ents.FindByClass("info_player_start")) do
-				if (v:GetPos().y == -64 or v:GetPos().y == -160 or v:GetPos().y == -256) then
-					table.insert(outsidespawns,v)
+			ply.PitDeath = false
+			if IsValid(outsidespawns) and #outsidespawns < 1 then

```

Comment:
IsValid is really more for entities. If you want to check that it is not nil just do if outsidespawns...